trio/_timeouts.py

```diff
@@ -92,7 +94,7 @@ class TooSlowError(Exception):
 
 
 @contextmanager
-def fail_at(deadline):
+def fail_at(deadline: float) -> Iterator[trio.CancelScope]:
```

Comment:
iterattor typehint makes this useless for autocomplete.

Reply:
`Iterator` is the correct type here, or alternatively `typing.Generator[trio.CancelScope, None, None]`. The function itself doesn't return a CM, it returns an iterator. `contextmanager` then transforms the function into one returning a CM. If the autocomplete tool doesn't have a special case for `contextmanager`, it's more a problem there...

Reply:
I don't care what tthe type checker says. I use type hints for autocomplete.

Reply:
What program are you using for autocomplete?
Could possibly define a different signature within `if TYPE_CHECKING` if it doesn't support decorators/`contextmanager` and is used widely enough to warrant special casing.

Reply:
I use PyCharm (relevant [bug](https://youtrack.jetbrains.com/issue/PY-36444/PyCharm-doesnt-infer-types-when-using-contextlib.contextmanager-decorator). Doing an ``if TYPE_CHECKING:`` and then having an empty function that returns ``AsyncContextManager`` instead would work out great.